functions/src/triggers/stages.triggers.ts

```diff
@@ -149,17 +149,47 @@ export const publishStageData = onDocumentWritten(
           .get();
         const publicData = publicDoc.data() as VoteForLeaderStagePublicData;
 
-        // Update participant rankings
         const participantRankings = publicData.participantRankings;
         participantRankings[participantPublicId] = data.rankings;
 
-        // TODO: When determining current leader,
-        // only use participants whose scores are in the top two
-        // for willingness to lead (WTL).
-        //
-        // Determine top two scores from WTL, then filter participantRankings
-        // by participants who have those scores.
-        const currentLeader = getCondorcetElectionWinner(participantRankings);
+        let currentLeader = getCondorcetElectionWinner(participantRankings);
+
+        // Search for the WTL stage ID
+        const stagesSnapshot = await app
+          .firestore()
+          .collection(`experiments/${experimentId}/stages`)
+          .get();
+        const wtlStageDoc = stagesSnapshot.docs
+          .map((doc) => doc.data())
+          .find((data) => data.kind === StageKind.WTLSurvey);
+
+        if (wtlStageDoc) {
+          const wtlDoc = await app
```

Comment:
Can you get the wtlDoc from the stagesSnapshot.docs above? Rather than calling Firestore again?

Reply:
I tried and it didn't quite work; we can look into this during the refactor.